src/lib.rs

```diff
@@ -1,35 +1,96 @@
 #![allow(unused_variables)] // TODO: remove this when more things are implemented
 
-wit_bindgen_guest_rust::import!("wit/wasi-clocks.wit.md");
-wit_bindgen_guest_rust::import!("wit/wasi-default-clocks.wit.md");
-wit_bindgen_guest_rust::import!("wit/wasi-filesystem.wit.md");
-wit_bindgen_guest_rust::import!("wit/wasi-logging.wit.md");
-wit_bindgen_guest_rust::import!("wit/wasi-poll.wit.md");
-wit_bindgen_guest_rust::import!("wit/wasi-random.wit.md");
-
-use std::arch::wasm32::unreachable;
-use std::ptr::null_mut;
+wit_bindgen_guest_rust::import!({
+    paths: [
+        "wit/wasi-clocks.wit.md",
+        "wit/wasi-default-clocks.wit.md",
+        "wit/wasi-filesystem.wit.md",
+        "wit/wasi-logging.wit.md",
+        "wit/wasi-poll.wit.md",
+        "wit/wasi-random.wit.md"
+    ],
+    raw_strings,
+    unchecked
+});
+
+use core::arch::wasm32::unreachable;
+use core::mem::forget;
+use core::ptr::null_mut;
+use core::slice;
 use wasi::*;
 
+extern crate alloc;
+
+// We're avoiding static initializers, so replace the standard assert macros
+// with simpler implementation.
+macro_rules! assert {
+    ($cond:expr $(,)?) => {
+        if !$cond {
+            unreachable()
+        }
+    };
+}
+macro_rules! assert_eq {
+    ($left:expr, $right:expr $(,)?) => {
+        assert!($left == $right);
+    };
+}
+
+// We're avoiding static initializers, so don't link in the default allocator.
+struct Alloc {}
+unsafe impl alloc::alloc::GlobalAlloc for Alloc {
+    unsafe fn alloc(&self, _: alloc::alloc::Layout) -> *mut u8 {
+        unreachable()
+    }
+    unsafe fn dealloc(&self, _: *mut u8, _: alloc::alloc::Layout) {
+        unreachable()
+    }
+}
+#[global_allocator]
+static ALLOC: Alloc = Alloc {};
```

Comment:
Is this necessary? It seems better to have a "static" assertion via CI where things break if `dlmalloc` is included rather than this which would lead to a runtime panic if this actually ends up being called.